pkg/cache/v2/delta.go

```diff
@@ -44,17 +45,22 @@ func respondDelta(request *DeltaRequest, value chan DeltaResponse, st *stream.St
 	return nil
 }
 
-func createDeltaResponse(request *DeltaRequest, st *stream.StreamState, resources map[string]types.Resource, systemVersion string) (*RawDeltaResponse, error) {
+func createDeltaResponse(request *DeltaRequest, st *stream.StreamState, resources map[string]types.ResourceWithTtl, systemVersion string) (*RawDeltaResponse, error) {
 	nextVersionMap := make(map[string]string)
-	filtered := make([]types.Resource, 0)
+	filtered := make([]types.ResourceWithTtl, 0)
 	toRemove := make([]string, 0)
 
 	// Wildcard can happen through CDS/LDS. If this is done we want to track all resources as well as track them in the version map
 	if st.IsWildcard {
-		for name, resource := range resources {
-			marshaledResource, err := MarshalResource(resource)
+		for name, r := range resources {
+			// Set the heartbeat here to false since I'm not entirely sure where to pull that from here?
+			maybeTtldResource, _, err := ttl.MaybeCreateTtlResourceIfSupported(r, name, request.TypeUrl, false)
```

Comment:
For heartbeating to work with Delta we need to be pushing delta updates here: https://github.com/envoyproxy/go-control-plane/blob/35430f0740d5573e1ef9bf38599b2eb4d4e71612/pkg/cache/v2/simple.go#L156

As mentioned earlier we can handle TTL + heartbeating in delta separately and make sure we add proper test coverage if you feel like that would be easier, no need to block the immediate delta work. 

If we are going to add this we should have test coverage for TTL and heartbeats for delta APIs

For all other responses not triggered by the heartbeat routine we should set false here

Reply:
Awesome, I have that written now just looking at the tests. I implemented the test for delta but it's getting thousands of responses which is obviously not write so something is messed up there